store/index.js

```diff
@@ -346,6 +393,46 @@ export const getters = {
         return p_state.dataDictionary.annotated[p_columnName].transformationHeuristic;
     },
 
+    getJsonOutput(p_state, p_getters) {
+
+        let jsonOutput = {};
+
+        // 1. Create a JSON output that contains annotations made thus far
+        Object.keys(p_state.dataDictionary.annotated).forEach(columnName => {
+
+            let columnOutput;
+
+            // A. If a column has been categorized/annotated, transform its annotations to json output format
+            if ( null !== p_state.columnToCategoryMap[columnName] ) {
```

Comment:
If I'm not mistaken `columnToCategoryMap` only contains mappings made on categorization page. We allow users to map a column to a category but don't enforce annotation of every column i.e., they can go past the annotation page and proceed to download page as long as they've annotated one column. So I think checking the `columnToCategoryMap` as the only condition will result in skipping columns that have been categorized but not annotated since there is no annotation for these columns. Not sure if that's what we want, what do you think?

Reply:
The logic currently is as follows:
```
if column has been categorized:
     - include in its data dictionary entry any non-Neurobagel contents and our 'Annotations' section
else:
    - include in its data dictionary entry any non-Neurobagel contents
```
When `initializeColumnToCategoryMap` is called by `processDataTable`, it initializes the `columnToCategoryMap` with all columns as keys and values as null. The reason `columnToCategoryMap` and categorization of a column is used here is because without categorization the column will not show up on the annotation page tabs. Thus we do know what at least had the possibility of being annotated.

> So I think checking the columnToCategoryMap as the only condition will result in skipping columns that have been categorized but not annotated since there is no annotation for these columns. Not sure if that's what we want, what do you think?

Maybe what you mean is: we can have empty annotations here since only one annotated column is needed to move to the download page? If so, I agree we can probably make this function not output a blank `Levels` and `Missing Values` for unannotated columns. Another related change would be to not include `Missing Values` if it is blank. This would be similar for continuous value columns and `Transformation`

Reply:
Thanks for the explanation, I forgot we keep empty annotations for unannotated columns. 

> Maybe what you mean is: we can have empty annotations here since only one annotated column is needed to move to the download page? If so, I agree we can probably make this function not output a blank Levels and Missing Values for unannotated, categorical columns. Another related change would be to not include MissingValues if it is blank. This would be similar for continuous value columns and Transformation and its MissingValues.

Yea, that's what I meant.

Reply:
Do you think the output should stay is as we should spec out changes in another issue/PR? Or make the change now not to output blank annotation entries?

Reply:
I'd say if it can be implemented in the `getJsonOutput` it's better to handle it here but if it requires changes in other parts of the code handle it separately in another issue/PR.

Reply:
To be discussed upon implementation of https://github.com/neurobagel/bagel-cli/issues/144

Reply:
@rmanaem I'm holding off on merge until that issue is complete then?

Reply:
No, I'd say let's get this out of the way and if there is any changes required we can open an issue and address them then.

Reply:
Any further changes to this code from new documentation will be addressed in a subsequent PR.